code/datums/extensions/cafe_prop.dm

```diff
@@ -0,0 +1,57 @@
+/datum/extension/cafe_prop
+	base_type = /datum/extension/cafe_prop
+	expected_type = /obj/item/chems
+
+/datum/extension/cafe_prop/proc/update_prop(var/obj/item/prop)
+	if(!istype(prop))
+		return
+
+	. = initial(prop.name)
+	if(prop.reagents?.total_volume > 0)
+		. = "[.] of"
+		var/decl/reagent/primary_reagent = prop.reagents.get_primary_reagent_decl()
+		if(primary_reagent.type != /decl/reagent/drink/ice && prop.reagents.has_reagent(/decl/reagent/drink/ice))
+			. = "[.] iced"
+		
+		if(istype(primary_reagent, /decl/reagent/drink/coffee))
+			
+			if(prop.reagents.has_reagent(/decl/reagent/nutriment/mint))
+				. = "[.] mint"
+			if(prop.reagents.has_reagent(/decl/reagent/drink/syrup_caramel))
```

Comment:
On BoH all syrups shared a parent type, `/datum/reagent/drink/syrup`, so perhaps that would be of use here. You could simply extract the name directly on all subtypes of `/decl/reagent/drink/syrup` instead of having all of these istypes.

Reply:
(*subtypes of* is important, because simple syrup, i.e. `/decl/reagent/drink/syrup` itself, is essentially identical to sugar when used in drinks)

Reply:
This is a bit out of scope to be honest - I would need to iterate the entire reagent list looking for syrup subtypes, and it would mess with the order of words (`mocha pumpkin spice caramel latte` feels off compared to `mocha caramel pumpkin spice latte`, I dunno).